staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -250,28 +265,27 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 			return err
 		}
 
-		params := generate.MakeParams(cmd, names)
 		name := info.Name
 		if len(name) > validation.DNS1035LabelMaxLength {
 			name = name[:validation.DNS1035LabelMaxLength]
 		}
-		params["default-name"] = name
+		o.DefaultName = name
 
 		// For objects that need a pod selector, derive it from the exposed object in case a user
 		// didn't explicitly specify one via --selector
-		if s, found := params["selector"]; found && generate.IsZero(s) {
+		if len(o.Selector) == 0 {
 			s, err := o.MapBasedSelectorForObject(info.Object)
 			if err != nil {
 				return cmdutil.UsageErrorf(cmd, "couldn't retrieve selectors via --selector flag or introspection: %v", err)
 			}
-			params["selector"] = s
+			o.Selector = s
 		}
 
-		isHeadlessService := params["cluster-ip"] == "None"
+		isHeadlessService := o.ClusterIP == "None"
```

Comment:
Reading through code, I see we never validated that, I'd probably add a check for empty or `None` value for this one.

Reply:
If you look at the test cases, https://github.com/kubernetes/kubectl/blob/1bf219da9bdd8c03758195f5b43f1ddfbe31c3e7/pkg/generate/versioned/service_test.go, cluster-ip can be any value, `""`, `10.10.10.10` or `None`